internal/eth/sendtx.go

```diff
@@ -18,88 +21,164 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/crypto/kzg4844"
 	"github.com/ethereum/go-ethereum/ethclient"
-	"github.com/ethereum/go-ethereum/log"
 	"github.com/holiman/uint256"
-	bb "github.com/primev/preconf_blob_bidder/core/mevcommit"
+	bb "github.com/primev/preconf_blob_bidder/internal/mevcommit"
+	"github.com/rs/zerolog/log"
 	"golang.org/x/exp/rand"
 )
 
+var defaultTimeout time.Duration
+
+// init initializes the defaultTimeout variable by reading the DEFAULT_TIMEOUT
+// environment variable. If not set or invalid, it defaults to 15 seconds.
+func init() {
+	timeoutStr := os.Getenv("DEFAULT_TIMEOUT")
+	if timeoutStr != "" {
+		timeoutSeconds, err := strconv.Atoi(timeoutStr)
+		if err != nil {
+			log.Warn().
+				Str("DEFAULT_TIMEOUT", timeoutStr).
+				Msg("Invalid DEFAULT_TIMEOUT value. Using default of 15 seconds.")
+			defaultTimeout = 15 * time.Second
+		} else {
+			defaultTimeout = time.Duration(timeoutSeconds) * time.Second
+			log.Info().
+				Dur("defaultTimeout", defaultTimeout).
+				Msg("defaultTimeout loaded from environment")
+		}
+	} else {
+		defaultTimeout = 15 * time.Second
+		log.Info().
+			Dur("defaultTimeout", defaultTimeout).
+			Msg("DEFAULT_TIMEOUT not set. Using default of 15 seconds.")
+	}
+}
 
+// SelfETHTransfer sends an ETH transfer transaction from the authenticated account.
 func SelfETHTransfer(client *ethclient.Client, authAcct bb.AuthAcct, value *big.Int, offset uint64) (*types.Transaction, uint64, error) {
+	// Set a timeout context
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
 	// Get the account's nonce
-	nonce, err := client.PendingNonceAt(context.Background(), authAcct.Address)
+	nonce, err := client.PendingNonceAt(ctx, authAcct.Address)
 	if err != nil {
+		log.Error().
+			Err(err).
+			Str("function", "PendingNonceAt").
+			Msg("Failed to get pending nonce")
 		return nil, 0, err
 	}
 
 	// Get the current base fee per gas from the latest block header
-	header, err := client.HeaderByNumber(context.Background(), nil)
+	header, err := client.HeaderByNumber(ctx, nil)
 	if err != nil {
+		log.Error().
+			Err(err).
+			Str("function", "HeaderByNumber").
+			Msg("Failed to get latest block header")
 		return nil, 0, err
 	}
-	baseFee := header.BaseFee
-
-	
-	blockNumber := header.Number.Uint64()
 
-	// Get the chain ID (this does not work with the Titan RPC)
-	chainID, err := client.NetworkID(context.Background())
+	// Get the chain ID
+	chainID, err := client.NetworkID(ctx)
 	if err != nil {
+		log.Error().
+			Err(err).
+			Str("function", "NetworkID").
+			Msg("Failed to get network ID")
 		return nil, 0, err
 	}
 
-	// Create a new EIP-1559 transaction
+	baseFee := header.BaseFee
+	blockNumber := header.Number.Uint64()
+
+	// Create a transaction with a priority fee.
+	priorityFee := big.NewInt(2_000_000_000) // 2 gwei in wei
```

Comment:
This two values can be passed directly to `types.DynamicFeeTx` without defining them here.

Reply:
`blockNumber` gets printed out later as a log. Is it better to pass `header.Number.Uint64()` value directly to both `types.DynamicFeeTx` and the log?

Similarly `priorityFee` value is used to calculate `maxFee`. Is it better to pass the maxFee directly into `types.DynamicFeeTx` as well?